SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -756,6 +756,7 @@ void CDiscAdjMultizoneDriver::SetObjFunction(unsigned short kind_recording) {
               ObjFunc += solver[FEA_SOL]->GetTotal_OFRefGeom()*Weight_ObjFunc;
               break;
             case TOPOL_COMPLIANCE:
+              static_cast<CFEASolver*>(solver[FEA_SOL])->Integrate_FSI_Loads(geometry, config);
               solver[FEA_SOL]->Compute_OFCompliance(geometry, solver, config);
               ObjFunc += solver[FEA_SOL]->GetTotal_OFCompliance()*Weight_ObjFunc;
```

Comment:
So FSI adjoints work, at least the derivatives wrt material properties are correct for reasonably complex cases.
One downside that should be addressed is that there is no easy way to introduce BGS relaxation, and that really limits the type of FSI cases we can solve.
The separate recording of the OF introduces the need to call some solver routines that are usually in the "compute path" before evaluating exotic functions that are computed in one zone but have indirect contributions from the other zones, this is probably not the norm but it needs to be kept in mind when adding new functions.